save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt

```diff
@@ -121,18 +123,25 @@ class App : ComponentWithScope<PropsWithChildren, AppState>() {
                 val responseText = text().await()
                 if (!ok || responseText == "null") null else Json.decodeFromString(responseText)
             }
+
+            val avatarNew: ImageInfo? = userInfoNew?.name?.let { name ->
+                get(
+                    "$apiUrl/users/$name/avatar",
+                    Headers(),
+                    loadingHandler = ::noopLoadingHandler,
+                )
+                    .unsafeMap {
+                        it.decodeFromJsonString<ImageInfo>()
+                    }
+            }
             userInfoNew?.let {
                 setState {
-                    userInfo = userInfoNew.copy(globalRole = globalRole)
+                    userInfo = userInfoNew.copy(globalRole = globalRole, avatar = avatarNew?.path)
```

Comment:
can we remove `avatar` on `userInfo` if we don't use it?

Reply:
We still decided to use the path to the avatar from the user object

Reply:
why we do it, if `User.avatar` is set on uploading a new avatar. No need to request it additionally

Reply:
No. In App.kt, we do not look for a user in the database, we only take the name in authentication

Reply:
I see, can we then fetch UserInfo from `/api/$v1/users` instead of fetching avatar info separately

Reply:
similar in `UserSettingsView.kt`, we request avatar and user info separately, but avatar is part of user info

Reply:
also, `/sec/user` returns only `userName`, but return object is `UserInfo` -- can we refactor it to String to avoid getting partially filled UserInfo object

Reply:
OK. Now I'm returning a string and not UserInfo, and I'm requesting a User, not an avatar.
I will change the UserSetting later as part of another issue